dot/core/messages.go

```diff
@@ -32,14 +32,15 @@ func (s *Service) validateTransaction(peerID peer.ID, head *types.Header, rt Run
 	externalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)}, tx...))
 	validity, err = rt.ValidateTransaction(externalExt)
 	if err != nil {
-		if errors.Is(err, runtime.ErrInvalidTransaction) {
+		logger.Debugf("failed to validate transaction: %s", err)
+		switch {
+		case errors.Is(err, runtimererrors.ErrInvalidTxn):
```

Comment:
Hmm could we have these errors defined in the runtime package perhaps? It seems rather strange to have `rt.ValidateTransaction` (from `runtime`) return errors defined in another package.

Reply:
To make sure I understand, remove the subpackage I made in runtime and just put those files back in runtime?

Reply:
Wellppp I guess? Alternatively you could have a `lib/runtime/transaction` package (well `internal/runtime/transaction`) and move your code in there, including `ValidateTransaction` (as `Validate` to avoid name cluttering). Maybe give this a quick try? So you would call `transaction.Validate(...)` and then check against `transaction.InvalidTransaction` etc. that would work right @timwu20 what do you think?

Reply:
Im thinking we should not make this change here and resolve it in https://github.com/ChainSafe/gossamer/issues/2849